### PR TITLE
Add number of exceptions icon to headings in a case and offences, display 'checkmark' icons when corrections made 

### PR DESCRIPTION
**Pull request by tausifpatel**
### Add number of exceptions icon to headings in a case and offences, display 'checkmark' icons that had corrections made 

User story - [BICAWS7-2473](https://dsdmoj.atlassian.net/jira/software/c/projects/BICAWS7/boards/1368?selectedIssue=BICAWS7-2473)

Changes:
- Display exception icons, indicating the number of exceptions on the tabs, in a case.
- Display checkmark icons on tabs where exceptions have been resolved.
- Display warning icons next to each offence that has exceptions to be resolved.
- Display checkmark icon next to each offence that has exceptions resolved.

Screenshot:
Screenshot 2024-03-26 at 10 26 39 am
Screenshot 2024-03-26 at 10 27 26 am

[BICAWS7-2473]: https://dsdmoj.atlassian.net/browse/BICAWS7-2473?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/features/CourtCaseDetails/Tabs/Panels/Offences/OffencesList/OffencesListRow.tsx [hunk 1]
```diff
@@ -2,15 +2,64 @@ import { Offence } from "@moj-bichard7-developers/bichard7-next-core/core/types/
...
 
 export const OffencesListRow = ({ offence, onClick }: OffencesListRowProps) => {
+  const courtCase = useCourtCase()
+  const exceptions = courtCase.aho.Exceptions
+  const updatedFields = getUpdatedFields(courtCase.aho, courtCase.updatedHearingOutcome)
+  const offenceAlerts = getOffenceAlertsDetails(exceptions, updatedFields)
+  const checkmarkIcon = (
+    <div className={"checkmark-icon"}>
+      <Image
+        key={offence.CourtOffenceSequenceNumber}
```
Comment (richrace) [inline src/features/CourtCaseDetails/Tabs/Panels/Offences/OffencesList/OffencesListRow.tsx:25]: This `key` attribute needs to be in the parent `div` to stop errors from the console

File: src/features/CourtCaseDetails/Tabs/Panels/Offences/OffencesList/OffencesListRow.tsx [hunk 2]
```diff
@@ -2,15 +2,64 @@ import { Offence } from "@moj-bichard7-developers/bichard7-next-core/core/types/
...
+        src={CHECKMARK_ICON_URL}
+        width={30}
+        height={30}
+        alt="Checkmark icon"
+      />
+    </div>
+  )
+  const warningIcon = (
+    <div className={"warning-icon"}>
+      <WarningIcon key={offence.CourtOffenceSequenceNumber} />
```
Comment (richrace) [inline src/features/CourtCaseDetails/Tabs/Panels/Offences/OffencesList/OffencesListRow.tsx:35]: This `key` attribute needs to be in the parent `div` to stop errors from the console

**Review by richrace — CHANGES_REQUESTED** [review]

File: src/features/CourtCaseDetails/Tabs/Panels/Offences/OffencesList/OffencesListRow.tsx (hunk 1 above)
Comment (tausifpatel) [reply src/features/CourtCaseDetails/Tabs/Panels/Offences/OffencesList/OffencesListRow.tsx]: Moved `key` attribute to parent `div`

File: src/features/CourtCaseDetails/Tabs/Panels/Offences/OffencesList/OffencesListRow.tsx (hunk 2 above)
Comment (tausifpatel) [reply src/features/CourtCaseDetails/Tabs/Panels/Offences/OffencesList/OffencesListRow.tsx]: Moved `key` attribute to parent `div`

**Review by richrace — DISMISSED** [review]

**Review by richrace — APPROVED** [review]

**Pull request merged**